contracts/contracts/strategies/AerodromeEthStrategy.sol

```diff
@@ -0,0 +1,592 @@
+        emit Withdrawal(address(oeth), address(lpTokenAddress), oethToBurn);
+    }
+
+    // Pool Peg keeping
+
+    /**
+     * @notice Mints and swaps the given oeth token amount for weth to maintain pool balance.
+     * @param _amountIn Amount of tokens to swap for.
+     * @param _minAmountOut Minimum amount of expected output amount (slippage adjusted)
+     * @param _tokenIn Address of the input token.
+     * @param _recipient Address of the recipient of weth tokens if we swap oeth for weth.
+     */
+    function swapAndRebalancePool(
+        uint256 _amountIn,
+        uint256 _minAmountOut,
+        address _tokenIn,
+        address _recipient
+    ) external onlyStrategist improvePoolBalance nonReentrant {
+        // tokenIn should be either oeth or weth.
+        require(
+            _tokenIn == address(weth) || _tokenIn == address(oeth),
+            "Invalid tokenIn"
+        );
+
+        IRouter.Route[] memory routes = new IRouter.Route[](1);
+
+        uint256 oethBalanceBefore;
+
+        if (_tokenIn == address(oeth)) {
+            IVault(vaultAddress).mintForStrategy(_amountIn);
+
+            IRouter.Route memory oethToWethRoute = IRouter.Route({
+                from: address(oeth),
+                to: address(weth),
+                stable: true,
+                factory: address(aeroFactoryAddress)
+            });
+
+            routes[0] = oethToWethRoute;
+        } else {
+            // Override recipient address so that the oeth can be burnt.
+            _recipient = address(this);
+            oethBalanceBefore = oeth.balanceOf(address(this));
+            IRouter.Route memory wethToOethRoute = IRouter.Route({
+                from: address(weth),
+                to: address(oeth),
+                stable: true,
+                factory: address(aeroFactoryAddress)
+            });
+
+            routes[0] = wethToOethRoute;
+        }
+
+        // Perform a swap to rebalance the pool
+        aeroRouterAddress.swapExactTokensForTokens(
+            _amountIn,
+            _minAmountOut,
+            routes,
+            _recipient,
+            block.timestamp
+        );
+
+        // If weth was swapped for oeth, burn the receieved oeth tokens.
+        if (_tokenIn == address(weth)) {
+            uint256 oethReceived = oeth.balanceOf(address(this)) -
+                oethBalanceBefore;
+            IVault(vaultAddress).burnForStrategy(oethReceived);
+        }
+    }
+
+    /**
+     * @notice Collect accumulated AERO rewards and send to the Harvester.
+     */
+    function collectRewardTokens()
+        external
+        override
+        onlyHarvester
+        nonReentrant
+    {
+        // Collect AERO
+        aeroGaugeAddress.getReward(address(this));
+        _collectRewardTokens();
+    }
+
+    function _lpWithdraw(uint256 _lpTokensAmount) internal {
+        // Claim remaining rwards before withdrawing LP Tokens
+        aeroGaugeAddress.getReward(address(this));
```

Comment:
Do we lose these reward tokens if we don't withdraw them? If not, it might just make things easier to not withdraw them until they are ready to be passed on. Less gas, and less things that could go wrong during a withdraw.

Reply:
Good point, looks like the gauge tracks the rewards during withdrawals. So no need to harvest during withdrawals. Removed.